designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiEndpoints.scala

```diff
@@ -0,0 +1,280 @@
+      )
+    )
+
+}
+
+object ScenarioActivityApiEndpoints {
+
+  object Dtos {
+    @derive(encoder, decoder, schema)
+    final case class ScenarioActivity private (comments: List[Comment], attachments: List[Attachment])
+
+    object ScenarioActivity {
+
+      def apply(activity: DbProcessActivity): ScenarioActivity =
+        new ScenarioActivity(
+          comments = activity.comments.map(Comment.apply),
+          attachments = activity.attachments.map(Attachment.apply)
+        )
+
+    }
+
+    @derive(encoder, decoder, schema)
+    final case class Comment private (
+        id: Long,
+        processVersionId: Long,
+        content: String,
+        user: String,
+        createDate: Instant
+    )
+
+    object Comment {
+
+      def apply(comment: DbComment): Comment =
+        new Comment(
+          id = comment.id,
+          processVersionId = comment.processVersionId.value,
+          content = comment.content,
+          user = comment.user,
+          createDate = comment.createDate
+        )
+
+    }
+
+    @derive(encoder, decoder, schema)
+    final case class Attachment private (
+        id: Long,
+        processVersionId: Long,
+        fileName: String,
+        user: String,
+        createDate: Instant
+    )
+
+    object Attachment {
+
+      def apply(attachment: DbAttachment): Attachment =
+        new Attachment(
+          id = attachment.id,
+          processVersionId = attachment.processVersionId.value,
+          fileName = attachment.fileName,
+          user = attachment.user,
+          createDate = attachment.createDate
+        )
+
+    }
+
+    final case class AddCommentRequest(scenarioName: ProcessName, versionId: VersionId, commentContent: String)
+
+    final case class DeleteCommentRequest(scenarioName: ProcessName, commentId: Long)
+
+    final case class AddAttachmentRequest(
+        scenarioName: ProcessName,
+        versionId: VersionId,
+        body: InputStream,
+        fileName: FileName
+    )
+
+    final case class GetAttachmentRequest(scenarioName: ProcessName, attachmentId: Long)
+
+    final case class GetAttachmentResponse(inputStream: InputStream, fileName: Option[String], contentType: String)
+
+    object GetAttachmentResponse {
+      val emptyResponse: GetAttachmentResponse =
+        GetAttachmentResponse(InputStream.nullInputStream(), None, MediaType.TextPlainUtf8.toString())
+    }
+
+    sealed trait ScenarioActivityErrors
+
+    object ScenarioActivityErrors {
+      case class NoScenario(scenarioName: ProcessName) extends ScenarioActivityErrors
+      case object NoPermission                         extends ScenarioActivityErrors with CustomAuthorizationError
+      case class NoComment(commentId: Long)            extends ScenarioActivityErrors
+
+      implicit val noScenarioCodec: Codec[String, NoScenario, CodecFormat.TextPlain] = {
+        Codec.string.map(
+          Mapping.from[String, NoScenario](_ => ???)(e => s"No scenario ${e.scenarioName} found")
```

Comment:
let's throw the Illegal State Exception with the meaningful message instead of leaving `???`